### ISSUE TITLE
syslog-ng-ctl reopen hangs after reload

### ISSUE THREAD

**Issue by Wernfried** [syslog-ng/syslog-ng#3422]
When you reload the config and try `reopen` afterwards then the process is hanging. syslog-ng seems to continue running normally but syslog-ng-ctl operation hangs.

```
$ /usr/sbin/syslog-ng-ctl reload --control=$MEDIATION_VAR/syslog-ng.src.ctl
OK Config reload initiated
$ /usr/sbin/syslog-ng-ctl reopen --control=$MEDIATION_VAR/syslog-ng.src.ctl
```
... hangs forever. Need to `kill -9` the process

In the past I faced hanging `syslog-ng-ctl reopen` occasionally, however I was not able to reproduce. Running reload reproduces the error reliable.

```
syslog-ng 3 (3.18.1)
Config version: 3.18
Installer-Version: 3.18.1
Revision: 
Compile-Date: Oct 17 2018 14:16:14
Module-Directory: //usr/lib64/syslog-ng
Module-Path: //usr/lib64/syslog-ng
Available-Modules: add-contextual-data,afamqp,affile,afprog,afsocket,afstomp,afuser,appmodel,basicfuncs,cef,confgen,cryptofuncs,csvparser,date,dbparser,disk-buffer,examples,graphite,hook-commands,json-plugin,kvformat,linux-kmsg-format,map-value-pairs,pseudofile,sdjournal,snmptrapd-parser,stardate,syslogformat,system-source,tags-parser,tfgetent,xml,afmongodb,afsql,http,afsmtp
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
Enable-Systemd: on

         Icon name: computer-server
           Chassis: server
  Operating System: Red Hat Enterprise Linux
       CPE OS Name: cpe:/o:redhat:enterprise_linux:7.5:GA:server
            Kernel: Linux 3.10.0-862.14.4.el7.x86_64
      Architecture: x86-64

```

Config file (partly)

```
options { 
   dns-cache-expire(86400); # 1 day
   dns-cache-size(1000);
   use-dns(persist_only);
   threaded(yes);
   dns-cache-hosts(/etc/hosts);
   stats-level(1); # default 0
   stats-freq(21600); # default 600
   use-uniqid(yes);
   flush-timeout(5000);
   mark-freq(86400); # default 1200
   keep-hostname(yes);
   time-zone("Europe/Zurich");
};

source s_internal { 
   internal(); 
};

filter f_statistics { level(info) and match("^Log statistics;" value("MESSAGE")) ; };
filter f_level_info { level(info..emerg); };
filter f_level_notice { level(notice..emerg); };
filter f_level_warning { level(warning..emerg); };

destination d_admin_mail {
   program("/opt/mediation/alert-mail.sh",
      template("${PID} ${MSG}\n") 
   );
};

destination d_internal_file {
   file("/var/log/mediation/syslog-ng.log", 
      perm(0640), owner(mediation), group(mediation),
      template("${ISODATE} ${LEVEL} [${PID}]: ${MSG}\n"),
      ts-format(iso),
      suppress(60)
   );   
};

destination d_internal_file_stats {
   file("/var/log/mediation/syslog-ng_stats.log", 
      perm(0640), owner(mediation), group(mediation),
      template("${ISODATE} [${PID}]: ${MSG}\n"),
      ts-format(iso),
      suppress(60)
   );   
};

log { 
   source(s_internal); 
   
   log {   
      filter(f_statistics);
      destination(d_internal_file_stats);
      flags(final); 
   }; 
   log {   
      destination(d_internal_file);
      
      filter(f_level_info); 
      # All messages above "info" go to internal logging
      destination(d_internal_syslogd);
      
      # All messages above "warning" are reported by mail
      filter(f_level_warning); 
      destination(d_admin_mail);
   }; 
   flags(flow-control); 
}; 

```

**Comment by gaborznagy** [comment]
Hi @Wernfried !

I've checked your bug report, and found that this bug has been fixed already by #2739 
I've tested both version (3.18.1 and our latest version 3.29.1) with your config and on 3.18 I was able to reproduce the issue, while on 3.29 not.

That path is released with version 3.22.

**Comment by Wernfried** [comment]
Ok, thank you for clarification.
Wernfried
